torch/_dynamo/eval_frame.py

```diff
@@ -754,6 +754,10 @@ def patch():
                 opt._cuda_graph_capture_health_check
             )
             opt.zero_grad = disable(opt.zero_grad)
+
+            if hasattr(opt, "_init_group"):
```

Comment:
Why do you prefer the far away interaction of "if this specific method is defined, X happens" vs just "Add @torch.dynamo.disable()" inside the optimizer code?

Reply:
yea, I'd actually also prefer just doing what Alban suggests, reducing action at a distance

Reply:
I agree, but right now there is a circular import with dynamo, let me see if I can extract disable into a separate module

Reply:
Yeah it would be tricky to do this, the disable function is tightly tied to the rest of dynamo and is non-trivial to isolate, so I'm going to continue to use TorchPatcher

Reply:
Why does Dynamo still needs to import optim though? If we specialize directly from here?

Reply:
I actually misspoke, I thought it was optim but it's not. either way here's the stack trace; it seems nontrivial to track down this circular import: 
https://gist.github.com/mlazos/ac886f0c18066a32002c695cb25f3dae